js/join.js

```diff
@@ -0,0 +1,305 @@
+const previewForFun = document.getElementById('previewForFun');
+const previewWhyRds = document.getElementById('previewWhyRds');
+const previewHeardAbout = document.getElementById('previewHeardAbout');
+const previous4 = document.getElementById('previous4');
+const submit = document.getElementById('next4');
+
+//Vatiables for Completed page
+const personalLink = document.getElementById('personalLink');
+const copyBtn = document.getElementById('copy');
+
+function fetchSavedDetails() {
+  fetch('https://api.realdevsquad.com/users/self', {
+    headers: { 'content-type': 'application/json' },
+    credentials: 'include',
+  })
+    .then((res) => res.json())
+    .then((res) => {
+      window.localStorage.setItem('firstName', res.first_name);
+      window.localStorage.setItem('lastName', res.last_name);
+      window.localStorage.setItem('userId', res.id);
+      const url = `https://api.realdevsquad.com/users/${localStorage.getItem(
+        'userId',
+      )}/intro`;
+      personalLink.innerText = url;
+    });
+}
+
+const pages = [page1, page2, page3, page4, page5, page6];
+
+function selectPage() {
+  let currentFlowState = window.localStorage.getItem('flowState');
+  showPage(currentFlowState);
+}
+
+function showPage(currentFlowState) {
+  for (let i = 0; i < pages.length; i++) {
+    if (i == currentFlowState) {
+      pages[i].classList.remove('hide-page');
+    } else {
+      pages[i].classList.add('hide-page');
+    }
+  }
+}
+
+//Validators
+function arePersonalDetailsValid() {
+  return (
+    state.value.trim().length > 3 &&
+    city.value.trim().length > 3 &&
+    country.value.trim().length > 3
+  );
+}
+
+function introPageChecker() {
+  return (
+    introduction.value.trim().split(' ').length > 100 &&
+    skills.value.trim().split(' ').length > 6 &&
+    college.value.trim().split(' ').length > 5 &&
+    forFun.value.trim().split(' ').length > 100 &&
+    funFact.value.trim().split(' ').length > 100
+  );
+}
+
+function whyRdsPageChecker() {
+  return whyRds.value.trim().split(' ').length > 100 && heardAbout.value != '';
+}
+
+function dataValidator(element, size) {
+  if (element.value.trim().split(' ').length > size) {
+    element.classList.add('correct-data');
+    element.classList.remove('incorrect-data');
+  } else {
+    element.classList.remove('correct-data');
+    element.classList.add('incorrect-data');
+  }
+}
+
+// Togglers Fillers
+function toggleNextButton() {
+  if (arePersonalDetailsValid()) {
+    next1.classList.remove('button-disabled');
+  } else {
+    next1.classList.add('button-disabled');
+  }
+  if (introPageChecker()) {
+    next2.classList.remove('button-disabled');
+  } else {
+    next2.classList.add('button-disabled');
+  }
+  if (whyRdsPageChecker()) {
+    previewBtn.classList.remove('button-disabled');
+  } else {
+    previewBtn.classList.add('button-disabled');
+  }
+}
+
+function autoFillTheFields() {
+  city.value = window.localStorage.getItem('city');
```

Comment:
can create an abstract functions like `addToLocalStorage` and `getFromLocalStorage`